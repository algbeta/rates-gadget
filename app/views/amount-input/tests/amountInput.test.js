import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AmountInput from '../amountInput';

describe('AmountInput tests', () => {
  test('AmountInput renders correctly', () => {
    const tree = renderer
      .create(
        <AmountInput
          inputExchangeAmount={() => {}}
          name="amountInput"
          amount={150}
          validationFailed={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Invalid AmountInput renders correctly', () => {
    const tree = renderer
      .create(
        <AmountInput
          inputExchangeAmount={() => {}}
          name="amountInput"
          amount={150}
          validationFailed
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('AmountInput handler test', () => {
    const handler = jest.fn();
    const wrapper = shallow(
      <AmountInput
        inputExchangeAmount={handler}
        name="amountInput"
        amount={150}
        validationFailed={false}
      />,
    );

    wrapper.find('input').simulate('change', { target: { value: '15' } });

    expect(handler).toHaveBeenCalledWith('15', 'amountInput');
  });
});
