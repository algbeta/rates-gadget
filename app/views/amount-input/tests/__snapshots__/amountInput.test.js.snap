// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInput tests AmountInput renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="amountInput"
  >
    Only numbers, - and . are allowed
  </label>
  <input
    className="form-control"
    name="amountInput"
    onChange={[Function]}
    value={150}
  />
</div>
`;

exports[`AmountInput tests Invalid AmountInput renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="amountInput"
  >
    Only numbers, - and . are allowed
  </label>
  <input
    className="form-control  is-invalid"
    name="amountInput"
    onChange={[Function]}
    value={150}
  />
</div>
`;
